l10n_it_central_journal_reportlab/wizard/print_giornale.py

```diff
@@ -0,0 +1,572 @@
+        ]
+
+        for line in self.env["account.move.line"].browse(move_line_ids):
+            start_row += 1
+            row = Paragraph(str(start_row), style_name)
+            date = Paragraph(format_date(self.env, line.date), style_name)
+            move_name = self._get_move_name_reportlab(
+                line.move_id.ref, line.move_id.name
+            )
+            move = Paragraph(move_name, style_name)
+            account_name = self._get_account_name_reportlab(
+                line.account_id.code, line.account_id.name
+            )
+            account = Paragraph(account_name, style_name)
+            if line.account_id.user_type_id.type in ["receivable", "payable"]:
+                name = Paragraph(str(line.partner_id.name or ""), style_name)
+            else:
+                name = Paragraph(str(line.name or ""), style_name)
+            debit = Paragraph(formatLang(self.env, line.debit), style_number)
+            credit = Paragraph(formatLang(self.env, line.credit), style_number)
+            list_balance.append((line.debit, line.credit))
+            line_data = [[row, date, move, account, name, debit, credit]]
+            if previous_move_name != move_name:
+                previous_move_name = move_name
+                tables.append(
+                    Table(line_data, colWidths=colwidths, style=style_table_line_above)
+                )
+            else:
+                tables.append(Table(line_data, colWidths=colwidths, style=style_table))
+        return tables, list_balance
+
+    def get_balance_data_report_giornale(
+            self, style_name, style_number, tot_debit, tot_credit, final=False):
+        if final:
+            name = Paragraph(_("Final Balance"), style_name)
+        else:
+            name = Paragraph(_("Balance"), style_name)
+
+        balance_data = [[
+            "", "", "", "",
+            name,
+            Paragraph(formatLang(self.env, tot_debit), style_number),
+            Paragraph(formatLang(self.env, tot_credit), style_number),
+        ]]
+        return balance_data
+
+    def create_report_giornale_reportlab(self):
+        pdf_bytes = io.BytesIO()
+
+        WIDTH, HEIGHT = A4
+        width_available = WIDTH - (2 * margin_left)
+        height_available = HEIGHT
+        report = canvas.Canvas(pdf_bytes, pagesize=A4)
+        height_available = \
+            self.get_template_header_report_giornale(report, height_available)
+
+        style_header = self.get_styles_report_giornale_line()["style_header"]
+        style_header_number = \
+            self.get_styles_report_giornale_line()["style_header_number"]
+        style_table = self.get_styles_report_giornale_line()["style_table"]
+        style_name = self.get_styles_report_giornale_line()["style_name"]
+        style_number = self.get_styles_report_giornale_line()["style_number"]
+        style_table_line_above = \
+            self.get_styles_report_giornale_line()["style_table_line_above"]
```

Comment:
Come mai viene chiamata diverse volte `get_styles_report_giornale_line`?
Non si potrebbe invece salvare in una variabile il suo risultato e poi usare quella?

(spin-off di https://github.com/OCA/l10n-italy/pull/2927#discussion_r1128181111)